atomix/cluster/src/main/java/io/atomix/raft/impl/DefaultRaftServer.java

```diff
@@ -107,12 +107,12 @@ public CompletableFuture<RaftServer> join(final Collection<MemberId> cluster) {
 
   @Override
   public CompletableFuture<RaftServer> leave() {
-    return context.leave().thenApply(v -> this);
+    return new ReconfigurationHelper(context).leave().thenApply(v -> this);
```

Comment:
:upside_down_face: Can probably be created only once? :shrug:

Reply:
There is no need to keep a long living object, right? These operations are not expected to be called frequently. join/leave will be called atmost once during the server's lifetime.